src/node/desktop/package.json

```diff
@@ -41,8 +41,9 @@
     "uuid": "^8.3.2"
   },
   "scripts": {
-    "build": "tsc",
+    "build": "tsc && yarn run copy-files",
     "clean": "ts-node scripts/clean.ts",
+    "copy-files": "ts-node scripts/copy-files.ts",
```

Comment:
Don't copy these files. Use them directly from source tree via relative location with `appState().resourcesPath().completePath()`; there are existing examples in the code for the static HTML files such as `html/error.html`. Get that working with dev builds, and it'll work with package builds (eventually, still working out some issues with Windows package builds).

Reply:
How should we handle widgets which want to depend on Typescript files? (Since after compilation, the associated Javascript will be placed in the `dist/` folder, rather than compiled in the same directory.)

Reply:
Need to look closer again, but my gut reaction is: the widgets themselves will have been compiled to javascript before they execute and thus be in the same (relative) dist location with any any files they are importing. I suspect there's an angle I'm missing with that "gut reaction" though.

For the webpacked version, everything in a given module will end up in a single JavaScript file. So all the main process code will be combined into one .js file, and all the renderer process code in another.

We do already use a tiny bit of code from the `core` library in the main preload script (something from `file-path.ts`). We can get away with that in a preload script since they can use node.js, currently. But renderer code outside the preload can't use it. There's also a small risk mentioned in #9737 that we might need to turn off most node.js even for the preload scripts.